### PR TITLE
Feature/function optimize vtest

### PR DESCRIPTION
**Pull request by darylccc**
**What does this PR do?**

**Why are these changes required?**

**This PR has been tested by:**
- Unit Tests
- Manual Testing

**Follow up**

**Extra details**

### CODE REVIEW

File: framework/src/main/resources/config.conf [hunk 1]
```diff
@@ -377,3 +377,7 @@ log.level = {
 eth = {
     gasPrice = 160000000000
 }
+debug.feature.effect = {
```
Comment (vchain007) [inline framework/src/main/resources/config.conf:380]: debug.feature.effect -> debug.feature.effectblocknum

File: framework/src/main/java/org/vision/core/config/args/Args.java [hunk 2]
```diff
@@ -738,6 +738,14 @@ public static void setParam(final String[] args, final String confFileName) {
             config.hasPath(Constant.ETH_GAS_PRICE) ? config
                     .getLong(Constant.ETH_GAS_PRICE) : 160_000_000_000L;
 
+    PARAMETER.spreadMintUnlimitedPledge =
```
Comment (vchain007) [inline framework/src/main/java/org/vision/core/config/args/Args.java:741]: spreadMintUnlimitedPledge -> spreadMintUnlimitedPledgeEffectBlocknum

File: framework/src/main/java/org/vision/core/config/args/Args.java [hunk 3]
```diff
@@ -738,6 +738,14 @@ public static void setParam(final String[] args, final String confFileName) {
...
+            config.hasPath(Constant.DEBUG_FEATURE_EFFECT_SPREADMINT_UNLIMITEDPLEDGE) ? config
+                    .getLong(Constant.DEBUG_FEATURE_EFFECT_SPREADMINT_UNLIMITEDPLEDGE) : 0L;
+
+    PARAMETER.witnessSort =
```
Comment (vchain007) [inline framework/src/main/java/org/vision/core/config/args/Args.java:745]: witnessSort -> witnessSortBlocknum

File: common/src/main/java/org/vision/core/Constant.java [hunk 4]
```diff
@@ -275,4 +275,6 @@
   public static final String HISTORY_BALANCE_LOOKUP = "storage.balance.history.lookup";
   public static final String LOCAL_HOST = "127.0.0.1";
   public static final String ETH_GAS_PRICE = "eth.gasPrice";
+  public static final String DEBUG_FEATURE_EFFECT_SPREADMINT_UNLIMITEDPLEDGE  = "debug.feature.effect.spreadMint.unlimitedPledge";
```
Comment (vchain007) [inline common/src/main/java/org/vision/core/Constant.java:278]: DEBUG_FEATURE_EFFECT_SPREADMINT_UNLIMITEDPLEDGE -> DEBUG_FEATURE_EFFECTBLOCKNUM_SPREADMINT_UNLIMITEDPLEDGE

debug.feature.effect.spreadMint.unlimitedPledge -> debug.feature.effectblocknum.spreadMint.unlimitedPledge

File: common/src/main/java/org/vision/core/Constant.java [hunk 5]
```diff
@@ -275,4 +275,6 @@
...
+  public static final String DEBUG_FEATURE_EFFECT_WITNESS_SORT  = "debug.feature.effect.witness.sort";
```
Comment (vchain007) [inline common/src/main/java/org/vision/core/Constant.java:279]: DEBUG_FEATURE_EFFECT_WITNESS_SORT -> DEBUG_FEATURE_EFFECTBLOCKNUM_WITNESS_SORT

debug.feature.effect.witness.sort -> debug.feature.effectblocknum.witness.sort

File: executor/src/main/java/org/vision/core/actuator/FreezeBalanceActuator.java [hunk 6]
```diff
@@ -248,10 +249,20 @@ public boolean validate() throws ContractValidateException {
       }
 
       if (spreadRelationShipCapsule != null){
-        if (spreadRelationShipCapsule.getExpireTimeForSpread() > dynamicStore.getLatestBlockHeaderTimestamp()){
-          throw new ContractValidateException("It's not time to re-freeze.");
+        if (isUnlimitedPledge){
+          if (!oldParent.equals(newParent) && spreadRelationShipCapsule.getExpireTimeForSpread() > dynamicStore.getLatestBlockHeaderTimestamp()){
+            throw new ContractValidateException("It's not time to modify parentAddress.");
```
Comment (vchain007) [inline executor/src/main/java/org/vision/core/actuator/FreezeBalanceActuator.java:254]: u 'd better add exact remaining time

File: executor/src/main/java/org/vision/core/actuator/FreezeBalanceActuator.java [hunk 7]
```diff
@@ -248,10 +249,20 @@ public boolean validate() throws ContractValidateException {
...
+          }
+        }else{
+          if (spreadRelationShipCapsule.getExpireTimeForSpread() > dynamicStore.getLatestBlockHeaderTimestamp()){
+            throw new ContractValidateException("It's not time to re-freeze.");
```
Comment (vchain007) [inline executor/src/main/java/org/vision/core/actuator/FreezeBalanceActuator.java:258]: u 'd better add exact remaining time

File: framework/src/main/java/org/vision/core/config/args/Args.java (hunk 3 above)
Comment (darylccc) [reply framework/src/main/java/org/vision/core/config/args/Args.java:745]: witnessSortEffectBlockNum

**Review by linusoutman — CHANGES_REQUESTED** [review]
approved

**Review by linusoutman — APPROVED** [review]

**Pull request merged**